frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/LogsTab.tsx

```diff
@@ -0,0 +1,217 @@
+  const podName = task.runDetails?.status.podName ?? '';
+  const { podContainers, selectedContainer, setSelectedContainer } = usePodContainerState(podName);
+  const [logs, loaded, error, refresh] = useFetchLogs(podName, selectedContainer?.name ?? '');
+  const [reRenderKey, setReRenderKey] = React.useState(0);
+  let data = 'No logs available';
+  if (logs) {
+    data = !loaded ? 'Loading...' : logs;
+  }
+  const refreshRate = 3000;
+  const maxLines = 500;
+
+  const [isPaused, setIsPaused] = React.useState(false);
+  const [isFullScreen] = React.useState(false);
+  const logViewerRef = React.useRef<{ scrollToBottom: () => void }>();
+
+  React.useEffect(() => {
+    const interval = setTimeout(() => {
+      if (isPaused) {
+        refresh();
+      }
+    }, refreshRate);
+
+    return () => clearInterval(interval);
+  }, [refresh, isPaused]);
+
+  React.useEffect(() => {
+    setReRenderKey((key) => key + 1);
+  }, [isHidden]);
+
+  React.useEffect(() => {
+    if (!isPaused) {
+      if (logViewerRef && logViewerRef.current) {
+        logViewerRef.current.scrollToBottom();
+      }
+    }
+  }, [isPaused]);
+
+  const onDownloadClick = () => {
+    const element = document.createElement('a');
+    const file = new Blob([data], { type: 'text/plain' });
+    element.href = URL.createObjectURL(file);
+    element.download = `${podName}-${selectedContainer}.txt`;
+    document.body.appendChild(element);
+    element.click();
+    document.body.removeChild(element);
+  };
+
+  const onScroll: React.ComponentProps<typeof LogViewer>['onScroll'] = ({
+    scrollOffsetToBottom,
+    scrollDirection,
+    scrollUpdateWasRequested,
+  }) => {
+    if (!scrollUpdateWasRequested) {
+      if (scrollOffsetToBottom < 1) {
+        setIsPaused(false);
+      } else if (scrollDirection === 'backward') {
+        setIsPaused(true);
+      }
+    }
+  };
+
+  return (
+    <>
+      {error ? (
+        <Alert
+          isExpandable
+          isInline
+          variant="danger"
+          title="An error occurred while retrieving the requested logs"
+          actionLinks={
+            <React.Fragment>
+              <AlertActionLink onClick={refresh}>Retry</AlertActionLink>
+            </React.Fragment>
+          }
+        >
+          <p>{error.message}</p>
+          <p>
+            <a onClick={refresh}>Retry</a>
+          </p>
+        </Alert>
+      ) : !loaded ? (
+        ''
+      ) : (
+        <Alert
+          isExpandable
+          isInline
+          variant="warning"
+          title="The log window displays partial content"
+        >
+          <p>
+            The log refreshes every {Math.floor(refreshRate / 1000)} seconds and displays the latest{' '}
+            {maxLines} lines. Exceptionally long lines are abridged. To view the full log for this
+            task, you can{' '}
+            <Button variant="link" isInline component="span" onClick={onDownloadClick}>
+              download the selected container logs
```

Comment:
changed `all container` to `selected container` as that's what we are supporting right now.